### ISSUE TITLE
Kettle should be an abstraction, not a boilerplate

### ISSUE THREAD

**Issue by danbovey** [JakeCobley/Kettle#56]
*TL;DR Kettle doesn't support a lot of use cases, so this enhancement could help.*

**Fork at: https://github.com/danbovey/Kettle**

With the current way Kettle is set up, the user would have to define different `styleDirectory`'s and rewrite `gulp.task` lines to get two CSS files in two places, say `build/css` and `build/css/microsite`.

So I propose we make Kettle an abstraction around gulp so that, rather than having a boilerplate gulpfile.js, we make a gulpfile.js framework that a user could write in ~30 seconds.

    // Pull in Kettle from npm
    var Kettle = require('Kettle');

    Kettle.scss({
        source: 'build/scss/app.scss',
        output: 'custom/css/style.css'
    });

This also hides away the `gulp-tasks` directory from the front-end project and makes updating Kettle easier as it would be a npm package.

**Comment by JakeCobley** [comment]
I'm interested by the idea, however this might conflict with the direction I have in mind for this project. I see you have already started work on this as a fork. I suggest when you have a strong base, submit a pull request and tag it in this discussion at which point we can consider the possibility of merging it into the project.

**Comment by dafzthomas** [comment]
This would definitely improve it as a tool by giving multiple options on how a developer can approach Kettle.

**Comment by JakeCobley** [comment]
I agree it would make it a better tool as a module you can just plugin in. But I'm not 100% sure it's what I want. Abstraction would take away the fact is an easy to edit standard format gulpfile which anyone can make their own.

Currently I see this project similar to bootstrap in the way you copy it and have full control over it, not really it's own application / module.

As I've said I will consider it upon @danbovey submitting a pull request.

**Comment by danbovey** [comment]
Cool stuff! It would make it more like the `bootstrap-sass` NPM module, where you define variables before importing bootstrap :smiley:

That's the standard way of doing it now, not overriding basic styles like primary button colour after the css is already imported. Same sort of thought process for Kettle hopefully.

**Comment by JakeCobley** [comment]
I understand its not how you'd edit bootstrap now, It was a bad example. I'm interested in applying kettle in this manner and have actually though of workarounds for possible future conflicts which were holding my judgment back.

I look forward to receiving your pull request which I will use to make the base.

**Comment by JakeCobley** [comment]
Closing this issue as It's not what the project is or where it's heading. I appreciate you contributions but in this case I will not be accepting any pull request.
